### ISSUE TITLE
Support environment package maps for non-relative names

### ISSUE THREAD

**Issue by guybedford** [systemjs/systemjs#937]
Currently - 
```javascript
System.config({
  packages: {
    'x': {
      map: {
        y: {
          browser: 'z'
        }
      }
    }
  }
});
```

is not supported due to the sync / async issues in the package normalization function for the `y` mapping to run during the prenormalize phase.

**Comment by guybedford** [comment]
Added in https://github.com/systemjs/systemjs/commit/cd39187ad86a04cd822b69b82965f7cd72720ba9.

**Comment by guybedford** [comment]
Released in 0.19.10.
